src/twitch/clip-downloader.ts

```
@@ -37,8 +37,7 @@ export class ClipDownloader extends EventEmitter {
 
         if (existsSync(appPath(mp4Path))) {
             logger.verbose(`Clip ${clip.title} found at ${appPath(mp4Path)}`);
-
-            return;
+            return Promise.reject(new Error('Clip already exist'));
```

You can throw the error directly, I think it's a bit cleaner